rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -124,23 +134,22 @@ void whenOldAndNewFederationArePresentReturnActiveFederationByActivationAge(
             currentBlockNumber--;
         }
 
-        // Arrange
-        Federation newFederation = getNewFakeFederation(newFedCreationBlockNumber);
-        Federation oldFederation = getNewFakeFederation(oldFedCreationBlockNumber);
-
-        FederationStorageProvider provider = mock(FederationStorageProvider.class);
-        when(provider.getNewFederation(federationConstants, activations)).thenReturn(newFederation);
-        when(provider.getOldFederation(federationConstants, activations)).thenReturn(oldFederation);
 
+        // Arrange
         Block executionBlock = mock(Block.class);
         when(executionBlock.getNumber()).thenReturn(currentBlockNumber);
 
-        FederationSupport federationSupport = new FederationSupportImpl(
-            federationConstants,
-            provider,
-            executionBlock,
-            activations
-        );
+        FederationSupport federationSupport = federationSupportBuilder
+            .withFederationConstants(federationConstants)
+            .withFederationStorageProvider(storageProvider)
+            .withRskExecutionBlock(executionBlock)
+            .withActivations(activations)
+            .build();
+
+        Federation newFederation = createFederation(newFedCreationBlockNumber);
+        Federation oldFederation = createFederation(oldFedCreationBlockNumber);
```

Comment:
This pretty much creates the same federation for both, so probably not very good idea for the purpose of the test

Reply:
yep, very good point